### DISCUSSION TITLE
[Proposal] Separate Nav.Item and Nav.MenuItem

### DISCUSSION THREAD

**Discussion by SevenOutman** [rsuite/rsuite#3059] [General]
## The Background

In [v5.13.0](https://github.com/rsuite/rsuite/releases/tag/v5.13.0), we discouraged usage of Dropdown within Nav for creating menus in a navigation bar. Instead, we introduced Nav.Menu API which renders a menu.

For "API simplicity", we didn't add a Nav.MenuItem component that renders menu items for Nav.Menu, but made Nav.Item render a menu item when it's used within a Nav.Menu. But it didn't turn out to be a good decision.

## The Problem

One noticeable problem is that a Nav.Item under Nav has different props than that under Nav.Menu, but it's impossible to give different type declarations depending on where Nav.Item is used.

```jsx
<Nav>
  <Nav.Item divider>Divider?</Nav.Item> // `divider` prop exists but does nothing when Nav.Item is not within Nav.Menu
  <Nav.Menu title="menu">
    <Nav.Item href="https://example.com">A link</Nav.Item> // `href` prop exists but does nothing when Nav.Item is within Nav.Menu
  </Nav.Menu>
</Nav>
```

The further reason behind the problem is that Nav.Item within vs. outside-of Nav.Menu **renders differently and behaves differently**. They don't *happen to* have different props, but are essentially different beings.

Similar issue exists with Nav.Menu rendering as direct child of Nav vs. within another Nav.Menu.

## The proposal

Add Nav.MenuItem component that's dedicated for usage within Nav.Menu.

```jsx
<Nav>
  <Nav.Item>Item</Nav.Item>
  <Nav.Menu title="menu">
    <Nav.MenuItem href="https://example.com">Menu item</Nav.MenuItem>
  </Nav.Menu>
</Nav>
```
## Referenece

Radix UI gives [a good example](https://www.radix-ui.com/docs/primitives/components/menubar) of not messing up with menu items at different levels.

**Comment by simonguo** [comment]
This actually returns to the [API design of Dropdown.Item](https://github.com/orgs/rsuite/discussions/2456
). The problem we essentially want to solve is that we don't want the Dropdown in Nav to mix with the Dropdown itself.

Splitting Nav.Item and Nav.MenuItem I think makes sense. However, it is necessary to consider the trouble caused by user upgrades. In the V5 version, it is best to keep the support for Nav.Item that can be nested in Nav.Menu.
